main/commons/src/main/java/org/cryptomator/common/mountpoint/MountPointChooser.java

```diff
@@ -138,20 +147,39 @@ default void cleanup(Path mountPoint) {
 	int getPriority();
 
 	/**
-	 * Called by the {@link MountPointChooserModule} to determine the execution order
-	 * of the registered MPCs. <b>Implementations usually should not override this
-	 * method.</b> This default implementation sorts the MPCs in ascending order
-	 * of their {@link #getPriority() priority.}<br>
+	 * Called by the {@link Volume} to determine the execution order of the registered MPCs.
+	 * <b>Implementations usually may not override this method.</b> This default implementation
+	 * sorts the MPCs in ascending order of their {@link #getPriority() priority.}<br>
 	 * <br>
 	 * <b>Original description:</b>
 	 * <p>{@inheritDoc}
 	 *
 	 * @implNote This default implementation sorts the MPCs in ascending order
-	 * of their {@link #getPriority() priority.}
+	 * of their {@link #getPriority() priority.} It depends on a correct implementation
+	 * of {@link #hashCode()} and {@link #equals(Object)} for the compared objects.
+	 * For this implementation "the least required correct implementation" maps
+	 * the following relationship correctly:<br>
+	 * If the objects {@code a} and {@code b} are {@code equal} as determined by
+	 * {@code #equals(Object)} then the {@code hashCodes} as determined by {@code #hashCode()}
+	 * of {@code a} and {@code b} must be equal. If the objects are {@code not equal}
+	 * then the {@code hashCodes} must be different.<br>
+	 * The default implementations of those two methods are sufficient.
+	 * @see Object#hashCode()
+	 * @see Object#equals(Object)
 	 */
 	@Override
 	default int compareTo(MountPointChooser other) {
-		//Sort by priority (ascending order)
-		return Integer.compare(this.getPriority(), other.getPriority());
+		Validate.notNull(other, "Other must not be null!");
+
+		//Try to compare by priority and sort in ascending order.
+		//1) Compare the priorities.
+		//2) If the result of the comparison is 0 because the priorities are equal,
+		//   we consider this a failure. In this case sort by hashCode (step 3).
+		//   Else return the result of the comparison.
+		//3) Return the result of the comparison of the hashCodes:
+		//   Either 0 (because they are equal and that's why they had the same priority)
+		//   or a non-null int (because they only share the priority).
+		int compResult = /* 1 */ Integer.compare(this.getPriority(), other.getPriority());
+		return compResult != 0 ? /* 2 */ compResult : /* 3 */ Integer.compare(this.hashCode(), other.hashCode());
```

Comment:
Drop the second degree sorting criteria. The hashcode is highly unstable and behaviour could change between versions. If two objects had the same priority, the sorting could still be correct "by mere chance". Flaws would be hidden.

Do not write code for the sole purpose to mask _potential mistakes_ in other parts of the program. If two MPCs had the same priority, **that** is either the code that needs to be corrected _or_ both are actually meant to be equally suited.

Therefore revert to the previous, trivial implementation.

Reply:
Resolved in personal conversation, as well as 6b6867b826f474e5926afb65c1ce2d3c8a58366d and 58e2ec4742d5e20e3fd361afe750539a67545a47.